web/components/TextBlock/TextBlock/TextBlock.tsx

```diff
@@ -41,18 +41,11 @@ const components: Partial<PortableTextComponents> = {
     italic: ({ children }) => <em>{children}</em>,
     underline: ({ children }) => <u>{children}</u>,
     code: ({ children }) => <code>{children}</code>,
-    internalLink: ({ text, value }) => (
-      /* href has been seen in a case where both link and internalLink marks applied */
-      <Link href={value?.href || getEntityPath(value.reference)}>{text}</Link>
-    ),
-    /* There are two different link marks, but currently TextBlock is used for
-     * everything. One variant is in blockContent (href), another in
-     * simpleBlockContent (internal/external). Would prefer to refactor the
-     * TextBlock approach, but it'd be a pretty big job.
-     */
     link: ({ children, value }) => {
-      const url =
-        value?.href || value?.internal?.slug?.current || value?.external;
+      const resolvedSlug = value?.internal?.slug?.current
+        ? getEntityPath(value.internal)
+        : null;
+      const url = resolvedSlug || value?.external;
```

Comment:
Nit: Maybe making `getEnittyPath` return `#` if `slug.current` is falsy was a mistake. :thinking: Very optional: rewrite `getEntityPath` to return something falsy for that case (Empty string? `null`? Not sure what's best), which would allow the following:

```suggestion
      const url = getEntityPath(value.internal) || value?.external;
```

(Not necessarily only to save these few bytes, but moreso because `#` isn't really a sensible default and using `getEntityPath` this way should be possible. But you might not want to scope this whole ordeal within the realm of this PR. Very, very optional comment.)

Reply:
Agreed. In order to limit the scope, I went with an empty string. This seems to be safe to pass to `next/link`. Using `null` feels more correct to me, but it's just a feeling and it would require checking that all other call sites could handle it (`next/link` errors out if you feed it `null` for `href`, it seems), so I just added a comment about it. 108c6cbe6bd082b77ca26924ba5fe316df25c76c